File: Java_stack/Bank Account/BankAccount.java
public class BankAccount {
    // MEMBER VARIABLES
    private double checkingBalance;
    private double savingsBalance;
    private String accountNumber;

    private static int accounts;
    private static double totalMoney; // refers to the sum of all bank account checking and savings balances

    // CONSTRUCTOR
    public BankAccount() {
        this.checkingBalance = 0;
        this.savingsBalance = 0;
        this.accountNumber = generateAccountNumber();
        accounts++;
    }
    
    // PRIVATE METHOD FOR GENERATING ACCOUNT NUMBER
    private String generateAccountNumber() {
        String accountNum = "";
        for (int i = 0; i < 10; i++) {
            accountNum += (int) (Math.random() * 10);
        }
        return accountNum;
    }

    // GETTERS
    public double getCheckingBalance() {
        return this.checkingBalance;
    }
    
    public double getSavingsBalance() {
        return this.savingsBalance;
    }
    
    public String getAccountNumber() {
        return this.accountNumber;
    }
    
    public static int getAccounts() {
        return accounts;
    }
    
    public static double getTotalMoney() {
        return totalMoney;
    }

    // METHODS
    // deposit
    public void deposit(String accountType, double amount) {
        if (amount > 0) {
            if (accountType.equalsIgnoreCase("checking")) {
                this.checkingBalance += amount;
                totalMoney += amount;
                System.out.println("Successfully deposited $" + amount + " to checking. New checking balance: $" + this.checkingBalance);
            } else if (accountType.equalsIgnoreCase("savings")) {
                this.savingsBalance += amount;
                totalMoney += amount;
                System.out.println("Successfully deposited $" + amount + " to savings. New savings balance: $" + this.savingsBalance);
            } else {
                System.out.println("Invalid account type. Please specify 'checking' or 'savings'.");
            }
        } else {
            System.out.println("Deposit amount must be positive.");
        }
    }
    
    // withdraw
    public void withdraw(String accountType, double amount) {
        if (amount <= 0) {
            System.out.println("Withdrawal amount must be positive.");
            return;
        }
        
        if (accountType.equalsIgnoreCase("checking")) {
            if (this.checkingBalance >= amount) {
                this.checkingBalance -= amount;
                totalMoney -= amount;
                System.out.println("Successfully withdrew $" + amount + " from checking. New checking balance: $" + this.checkingBalance);
            } else {
                System.out.println("Insufficient funds in checking account.");
            }
        } else if (accountType.equalsIgnoreCase("savings")) {
            if (this.savingsBalance >= amount) {
                this.savingsBalance -= amount;
                totalMoney -= amount;
                System.out.println("Successfully withdrew $" + amount + " from savings. New savings balance: $" + this.savingsBalance);
            } else {
                System.out.println("Insufficient funds in savings account.");
            }
        } else {
            System.out.println("Invalid account type. Please specify 'checking' or 'savings'.");
        }
    }
    
    // getBalance
    public void getBalance() {
        System.out.println("Account #" + this.accountNumber);
        System.out.println("Checking Balance: $" + this.checkingBalance);
        System.out.println("Savings Balance: $" + this.savingsBalance);
        System.out.println("Total Balance: $" + (this.checkingBalance + this.savingsBalance));
    }
}
    

File: Python_Stack/BankAccount/BankAccount.py
class BankAccount:
    def __init__(self, interest_rate=0.01, balance=0):
        self.interest_rate = interest_rate
        self.balance = balance 

    def deposit(self, amount):
        self.balance += amount
        print(f"Deposited ${amount}")
        return self

    def withdraw(self, amount):
        if self.balance >= amount:
            self.balance -= amount
            print(f"Withdrew ${amount}")
        else:
            print("Insufficient Balance")
        return self

    def display_account_info(self):
        print(f"Balance: ${self.balance:.2f}")
        return self

    def yield_interest_rate(self):
        if self.balance > 0:
            interest = self.balance * self.interest_rate
            self.balance += interest
            print(f"Interest Added: ${interest:.2f}")
        else:
            print("No interest applied (balance is zero)")
        return self


account1 = BankAccount()
account2 = BankAccount()

account1.deposit(400).deposit(500).deposit(200).withdraw(600).yield_interest_rate(). display_account_info()
account2.deposit(800).deposit(15000).withdraw(200).withdraw(100).withdraw(50).withdraw(150).yield_interest_rate().display_account_info()     